### PR TITLE
Add get_array method

### PR DESCRIPTION
**Pull request by okuta**
Rel #2493

### CODE REVIEW

File: chainer/variable.py [hunk 1]
```diff
@@ -529,3 +529,22 @@ def __hash__(self):
...
+
+def get_array(arg):
+    """Returns ndarray data.
+
+    This method is helper function for :class:`~chainer.Variable`.
+
+    Args:
+        arg (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            If `arg` type is :class:`chainer.Variable`, it returns `arg.data`.
```
Comment (unnonouno) [inline chainer/variable.py]: Use two backquotes.

File: tests/chainer_tests/test_variable.py [hunk 2]
```diff
@@ -839,4 +839,16 @@ def test_backward_gpu(self):
         self.check_backward(cuda.to_gpu(self.x))
 
 
+class TestGetArray(unittest.TestCase):
+
+    def test_variable(self):
+        x = np.zeros((10,))
+        v = chainer.Variable(x)
+        self.assertIsInstance(chainer.get_array(v), x)
```
Comment (unnonouno) [inline tests/chainer_tests/test_variable.py]: `numpy.ndarray`?

File: tests/chainer_tests/test_variable.py [hunk 3]
```diff
@@ -839,4 +839,16 @@ def test_backward_gpu(self):
...
+
+    def test_ndarray(self):
+        x = np.zeros((10,))
+        self.assertIsInstance(chainer.get_array(x), x)
```
Comment (unnonouno) [inline tests/chainer_tests/test_variable.py]: ditto

File: tests/chainer_tests/test_variable.py [hunk 4]
```diff
@@ -839,4 +839,16 @@ def test_backward_gpu(self):
...
 
 
+class TestGetArray(unittest.TestCase):
+
+    def test_variable(self):
+        x = np.zeros((10,))
+        v = chainer.Variable(x)
+        self.assertIsInstance(chainer.get_array(v), x)
+
+    def test_ndarray(self):
```
Comment (unnonouno) [inline tests/chainer_tests/test_variable.py]: Add test for `cupy.ndarray`.

File: chainer/variable.py [hunk 5]
```diff
@@ -529,3 +529,22 @@ def __hash__(self):
         return super(Variable, self).__hash__()
 
     __array_priority__ = 200
+
+
+def get_array(arg):
+    """Returns ndarray data.
+
+    This method is helper function for :class:`~chainer.Variable`.
```
Comment (unnonouno) [inline chainer/variable.py]: Write a description about this method like
"It returns an ndarray which a given variable holds."

**Comment by unnonouno** [comment]
And fix rst.

File: chainer/variable.py (hunk 5 above)
Comment (unnonouno) [inline chainer/variable.py]: And write its usecase.

Comment (beam2d) [inline chainer/variable.py]: Write the motivation behind this API, too. In particular, write the difference from just writing `x.data` to extract the array.

File: chainer/variable.py [hunk 6]
```diff
@@ -529,3 +529,22 @@ def __hash__(self):
         return super(Variable, self).__hash__()
 
     __array_priority__ = 200
+
+
+def get_array(arg):
+    """Returns ndarray data.
```
Comment (beam2d) [inline chainer/variable.py]: I think what "ndarray data" means is difficult to understand for beginners. How about writing as follows:
```
"""Extracts the data array of a :class:`~chainer.Variable`.
```
(I think careful readers understand the spec by reading the argument docs, but it's better to make any part of the document easy to understand and difficult to misunderstand.)

File: chainer/variable.py [hunk 7]
```diff
@@ -529,3 +529,22 @@ def __hash__(self):
...
+
+    This method is helper function for :class:`~chainer.Variable`.
+
+    Args:
+        arg (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            If ``arg`` type is :class:`chainer.Variable`, it returns
+            ``arg.data``. Otherwise it returns ``arg``.
```
Comment (beam2d) [inline chainer/variable.py]: This is a description of the returned value, and so it should be placed in the "Returns" section. Write the explanation of the argument itself here, like `"Object to extract an array from."`.

**Comment by okuta** [comment]
This PR is no longer needed.

**Pull request closed without merging**